lua/nvim-treesitter/info.lua

```diff
@@ -21,57 +21,113 @@ local function install_info()
+-- ->
+-- { default = {'mod1', 'mod3'}, mod2 = {'sub1', 'sub2'}}
+local function namespace_modules(modulelist)
+  local modules = {}
+  for _, module in ipairs(modulelist) do
+    if module:find('%.') then
+      local namespace, submodule = module:match('^(.*)%.(.*)$')
+      if not modules[namespace] then
+        modules[namespace] = {}
+      end
+      table.insert(modules[namespace], submodule)
     else
-      api.nvim_out_write('✗')
+      if not modules.default then
+        modules.default = {}
+      end
+      table.insert(modules.default, module)
     end
-    api.nvim_out_write('\n')
   end
+  return modules
 end
 
-local function print_info_modules(parserlist)
-  table.sort(parserlist, function(a, b) return #a > #b end)
-  local max_str_len = #parserlist[1]
-
-  local header = string.rep(' ', max_str_len + 1)
-  local mods = configs.available_modules()
-  table.sort(mods)
-  for _, mod in pairs(mods) do
-    header = string.format('%s %s ', header, mod)
+local function longest_string_length(list)
+  local length = 0
+  for _, value in ipairs(list) do
+    if #value > length then
+      length = #value
+    end
   end
-  api.nvim_out_write(header .. '\n')
+  return length
+end
 
-  table.sort(parserlist)
-  for _, lang in pairs(parserlist) do
-    local padding = string.rep(' ', max_str_len - #lang + 1)
-    api.nvim_out_write(lang .. ":" .. padding)
+local function append_module_table(curbuf, parserlist, namespace, modulelist)
+  local maxlen_parser = longest_string_length(parserlist)
+  table.sort(modulelist)
+
+  -- header
+  local header = '>> ' .. namespace .. string.rep(' ', maxlen_parser - #namespace - 1)
+  for _, module in pairs(modulelist) do
+    header = header .. module .. '  '
+  end
+  api.nvim_buf_set_lines(curbuf, -1, -1, true, {header})
 
-    for _, mod in pairs(mods) do
-      if configs.is_enabled(mod, lang) then
-        api.nvim_out_write('✓')
+  -- actual table
+  for _, parser in ipairs(parserlist) do
+    local padding = string.rep(' ', maxlen_parser - #parser + 2)
+    local line = parser ..  padding
+    local namespace_prefix = (namespace == 'default') and '' or namespace .. '.'
+    for _, module in pairs(modulelist) do
+      local modlen = #module
+      module = namespace_prefix .. module
+      if configs.is_enabled(module, parser) then
+        line = line .. '✓'
       else
-        api.nvim_out_write('✗')
+        line = line .. '✗'
       end
-      api.nvim_out_write(string.rep(' ', #mod + 1))
+      line = line .. string.rep(' ', modlen + 1)
     end
-    api.nvim_out_write('\n')
+    api.nvim_buf_set_lines(curbuf, -1, -1, true, {line})
   end
+
+  api.nvim_buf_set_lines(curbuf, -1, -1, true, {''})
+end
+
+local function print_info_modules(parserlist, module)
+  api.nvim_command('enew')
+  local curbuf = api.nvim_get_current_buf()
+
+  local modules
+  if module then
+    modules = namespace_modules({module})
+  else
+    modules = namespace_modules(configs.available_modules())
+  end
```

Comment:
cool cool.
a little lua trick here:
```lua
local modules = module and { module } or config.availabe_modules
```
works more like a ternary but if/else is fine too :)

Reply:
Oh, right. I even used it [here](https://github.com/nvim-treesitter/nvim-treesitter/pull/1143/files#diff-971a54c46b5207639b94c6a7603138567fcbaa268e87439eab2723371bea2519R72).

Still getting used to Lua and its idoms and best practices. :)